bin/server.js

```diff
@@ -10,189 +10,210 @@ var exchanges     = require('../auth/exchanges');
 var ScopeResolver = require('../auth/scoperesolver');
 var taskcluster   = require('taskcluster-client');
 var url           = require('url');
-
-
-/** Launch server */
-var launch = async function(profile) {
-  // Load configuration
-  var cfg = base.config({
-    defaults:     require('../config/defaults'),
-    profile:      require('../config/' + profile),
-    envs: [
-      'auth_tableSigningKey',
-      'auth_tableCryptoKey',
-      'auth_publishMetaData',
-      'server_publicUrl',
-      'server_cookieSecret',
-      'azure_accountName',
-      'azure_accountKey',
-      'pulse_username',
-      'pulse_password',
-      'aws_accessKeyId',
-      'aws_secretAccessKey',
-      'influx_connectionString',
-      'auth_rootAccessToken',
-      'auth_azureAccounts'
-    ],
-    filename:     'taskcluster-auth'
-  });
-
-  // Create InfluxDB connection for submitting statistics
-  var influx = new base.stats.Influx({
-    connectionString:   cfg.get('influx:connectionString'),
-    maxDelay:           cfg.get('influx:maxDelay'),
-    maxPendingPoints:   cfg.get('influx:maxPendingPoints')
-  });
-
-  // Start monitoring the process
-  base.stats.startProcessUsageReporting({
-    drain:      influx,
-    component:  cfg.get('auth:statsComponent'),
-    process:    'server'
-  });
-
-  // Create resolver
-  var resolver = new ScopeResolver({
-    maxLastUsedDelay: cfg.get('auth:maxLastUsedDelay'),
-  });
-
-  // Configure client table
-  var Client = data.Client.setup({
-    table:        cfg.get('auth:clientTableName'),
-    credentials:  cfg.get('azure'),
-    signingKey:   cfg.get('auth:tableSigningKey'),
-    cryptoKey:    cfg.get('auth:tableCryptoKey'),
-    drain:        influx,
-    component:    cfg.get('auth:statsComponent'),
-    process:      'server',
-    context:      {resolver}
-  });
-
-  // Configure role table
-  var Role = data.Role.setup({
-    table:        cfg.get('auth:rolesTableName'),
-    credentials:  cfg.get('azure'),
-    signingKey:   cfg.get('auth:tableSigningKey'),
-    drain:        influx,
-    component:    cfg.get('auth:statsComponent'),
-    process:      'server',
-    context:      {resolver}
-  });
-
-  // Initialize validator and publish schemas if needed
-  var validator, publisher;
-  await Promise.all([
-    (async () => {
-      validator = await base.validator({
+var loader        = require('taskcluster-lib-loader');
+var app           = require('taskcluster-lib-app');
+
+// Create component loader
+let load = loader({
+  cfg: {
+    requires: ['profile'],
+    setup: ({profile}) => {
+      return base.config({
```

Comment:
Sure, but it's not necessary here.  Well, in a subsequent patch, query string support might be useful (for listClients), but not in this PR.